📦 Improve docs on PDF exports for downloads (#1966)
jupyter-book/mystmd: docs/website-downloads.md

## Add a download link

Each download link entry has configuration that modifies its behavior.
Note that each entry may only specify one of `id`, `file`, or `url`.
Descriptions of these fields and other available fields are in the table below from the [downloads configuration](#frontmatter:downloads).

![](#table-frontmatter-downloads)

For example,

```{code-block} yaml
:label: config-page-download
:filename: article.md
---
downloads:
  - file: ./interesting-photo.png
    title: An interesting photograph
---
```

:::{tip} Define downloads for all pages
In [the above example](#config-page-download), the download configuration was defined for a single page (article.md). If you want to add downloads to every page, you can set the download information at the project level:

```{code-block} yaml
:label: config-proj-download
:filename: myst.yml

project:
  ...
  downloads:
    - file: ./interesting-photo.png
      title: An interesting photograph

```

:::

## Include an exported PDF

If you want to include a PDF of your document with the downloads, take these steps:

> 1. **Create a PDF export target**. For examples of how to build PDFs, see [](creating-pdf-documents.md). Let's say the PDF was output to `./mydoc.pdf`, then the following frontmatter would define a PDF export and give it a unique identifier `my-document-export`:
>    ```{code-block} yaml
>    :filename: article.md
>    :linenos:
>    :emphasize-lines: 3,4,5,6
>    ---
>    exports:

Reviewer: ```suggestion
1. **Create a PDF export target**. For example, the following page frontmatter defines a PDF export, gives it the unique identifier `my-document-export`, and will output the file `exports/my-document.pdf`:
   ```{code-block} yaml
   :filename: article.md
   :linenos:
   :emphasize-lines: 3,4,5,6
   ---
   exports:
```

     - format: pdf
       template: lapreprint-typst
       output: exports/my-document.pdf
       id: my-document-export
   ---
   ```
2. **Add a download for that export**. The `id` field should match the one defined for your PDF, e.g.
   ```{code-block} yaml
   :filename: article.md
   :linenos:
   :emphasize-lines: 8,9
   ---
   exports:
     - format: pdf
       template: lapreprint-typst
       output: exports/my-document.pdf
       id: my-document-export
   downloads:
     - id: my-document-export
       title: A PDF of this document
   ---
   ```
3. **Build the PDF**. Run the `myst build` command to build the PDF, e.g.
   ```bash
   myst build --pdf
   ```

> 
> An entry for this PDF will now show up in your page's downloads dropdown.
> 
> :::{note} Updating your GitHub Pages deployment
> :class: dropdown
> 

Reviewer: ```suggestion
4. **Build your website**. Now that you've built the PDF and added frontmatter for the download button, re-building your site will add a new download dropdown linked to the PDF that you've exported.

### Including exported files with GitHub Pages
```

> If you're [deploying a static site with GitHub pages](./deployment-github-pages.md), then you will need to add a new step that builds the PDF _before_ building the website. For example:
> 
> ```{code-block} yaml
> :filename: .github/workflows/deploy.yml

Reviewer: ````suggestion
If you're [deploying a static site with GitHub pages](./deployment-github-pages.md), then you will need _two build steps_ to add exported PDF files to your website. Ensure your content [has the proper PDF export frontmatter](#include-exported-pdf), then follow these two steps in your CI.

1. First, install the PDF build dependencies and build the PDF with `myst build --pdf`. In the example below, we'll show how to install Typst with the [`setup-typst` GitHub action][typst-gha].
2. Second, build the website with `myst build --html`. Because your PDF has already been generated, your website will now include it.

See below for sample configuration that accomplishes this:

:::{note} Example GitHub Action configuration to include PDF with GitHub Pages
:class: dropdown
```{code-block} yaml
:filename: .github/workflows/deploy.yml
````

:emphasize-lines: 17, 18
:linenos:

jobs:
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    steps:

>       - uses: actions/checkout@v4
>       - name: Setup Pages
>         uses: actions/configure-pages@v3
>       - uses: actions/setup-node@v4
>         with:
>           node-version: 18.x
>       - name: Install MyST Markdown
>         run: npm install -g mystmd
>       - name: Build PDF Assets
>         run: myst build --pdf
>       - name: Build HTML Assets
>         run: myst build --html
>       - name: Upload artifact
>         uses: actions/upload-pages-artifact@v3
>         with:
>           path: './_build/html'
>       - name: Deploy to GitHub Pages
>         id: deployment
>         uses: actions/deploy-pages@v4

Reviewer: ```suggestion
      - uses: actions/checkout@v4
      - name: Setup Pages
        uses: actions/configure-pages@v3
      - uses: actions/setup-node@v4
        with:
          node-version: 18.x
      - name: Install MyST Markdown
        run: npm install -g mystmd
      - name: Setup Typst
        uses: typst-community/setup-typst@v4        
      - name: Build PDF Assets
        run: myst build --pdf
      - name: Build HTML Assets
        run: myst build --html
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v3
        with:
          path: './_build/html'
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v4
```

```

> You will also need to ensure that the PDF typesetter (e.g. Typst) is installed in your CI environment. A useful tool is the [`setup-typst` GitHub action][typst-gha], which automates this:

Reviewer: ```suggestion
```

1. **Create a PDF export target**. For example, the following page frontmatter defines a PDF export, gives it the unique identifier `my-document-export`, and will output the file `exports/my-document.pdf`:
   ```{code-block} yaml
   :filename: article.md
   :linenos:
   :emphasize-lines: 3,4,5,6
   ---
   exports:
     - format: pdf
       template: lapreprint-typst
       output: exports/my-document.pdf
       id: my-document-export
   ---
   ```
2. **Add a download for that export**. The `id` field should match the one defined for your PDF, e.g.
   ```{code-block} yaml
   :filename: article.md
   :linenos:
   :emphasize-lines: 8,9
   ---
   exports:
     - format: pdf
       template: lapreprint-typst
       output: exports/my-document.pdf
       id: my-document-export
   downloads:
     - id: my-document-export
       title: A PDF of this document
   ---
   ```
3. **Build the PDF**. Run the `myst build` command to build the PDF, e.g.
   ```bash
   myst build --pdf
   ```
4. **Build your website**. Now that you've built the PDF and added frontmatter for the download button, re-building your site will add a new download dropdown linked to the PDF that you've exported.

> ### Including exported files with GitHub Pages

Reviewer: ```suggestion
### Include exported files with GitHub Pages
```
